packages/cli/src/checks/INTERNAL_MISMATCH.ts

```diff
@@ -22,9 +22,11 @@ export default makeCheck<ErrorType>({
         for (let depName in deps) {
           let range = deps[depName];
           let dependencyWorkspace = allWorkspaces.get(depName);
+          
           if (
             dependencyWorkspace !== undefined &&
-            !semver.satisfies(dependencyWorkspace.packageJson.version, range)
+            !semver.satisfies(dependencyWorkspace.packageJson.version, range) && 
+            !range.startsWith("npm:")
```

Comment:
```suggestion
            !range.startsWith("npm:") &&
            !semver.satisfies(dependencyWorkspace.packageJson.version, range)
```

Reply:
I know about the existence of `npm:` ranges and used them a couple of times... but damn, I can't find any docs that would mention them 😅 

In general, we likely should relax this check more and always allow things like `file:`, `link:`, etc. We don't necessarily have to do it here but maybe a simple regex that would allow any protocol would do the job here? cc @emmatown do you have any opinions about this one?

Reply:
It's mentioned here: https://github.com/Thinkmill/manypkg?tab=readme-ov-file#ignoring-this-rule

Reply:
I did the change you requested here

Reply:
I meant that I can't find this in the npm/yarn/pnpm docs. pnpm has a mention of this in one of examples but it doesn't even explain what it does 😅

Reply:
> In general, we likely should relax this check more and always allow things like file:, link:, etc. We don't necessarily have to do it here but maybe a simple regex that would allow any protocol would do the job here?

This sounds good to me